modules/tinymce/src/core/test/ts/browser/bookmark/BookmarksTest.ts

```diff
@@ -236,6 +236,27 @@ describe('browser.tinymce.core.bookmark.BookmarksTest', () => {
     TinyAssertions.assertCursor(editor, [ 0, 0 ], 1);
   }));
 
+  it('TINY-9194: should shift caret into zwnbsp if the bookmark is right after the cef element', bookmarkTest((editor) => {
```

Comment:
Maybe also stick in a test in SelectionOverridesTest.ts where it setups up the editor with a inline noneditable span. Places the caret after or before the zwnbsp characters then it should lean into the zwnbsp character. So for `<p><span contenteditable="false">CEF</span><zwnbsp text node></p>` it goes from `[0], 2` to `[0, 1], 1` for example.